c7n/resources/ebs.py

```diff
@@ -52,6 +52,47 @@ class SnapshotAge(AgeFilter):
     date_attribute = 'StartTime'
 
 
+####This filter will either return resources that are NOT AMI snapshots (value=False)
+####OR
+####it will return ONLY those that are AMI snapshots (value=True)
+@Snapshot.filter_registry.register('ami-snapshot')
+class SnapshotAmiSnapshot(Filter):
+    
+    schema = type_schema('ami-snapshot', value={'type': 'boolean'})
+    
+    def validate(self):
+        if self.data.get('value' != True or False):
+            raise FilterValidationError(
+                "invalid config: expected boolean value")
+        return self
+    
+    #get all snapshots tied to an AMI
+    def get_ami_snapshots(self):
+        c = local_session(self.manager.session_factory).client('ec2')
+        amis = []
+        for i in c.describe_images(Owners=['self'])['Images']:
+            for dev in i.get('BlockDeviceMappings'):
+                if 'Ebs' in dev and 'SnapshotId' in dev['Ebs']:
+                    amis.append(dev['Ebs']['SnapshotId'])
+        return amis
+    
+    #filter resources against the list of ami snapshots
+    def process(self, snapshots, event=None):
+        matches = []
+        ami = self.get_ami_snapshots()
+        if self.data.get('value', True):
```

Comment:
you can avoid this duplication, by using the value to define the operator that will be used.

op = self.data.get('value', True) and operator.contains or lambda a, b: b not in a